zero/src/block_interval.rs

```diff
@@ -31,75 +29,51 @@ pub enum BlockInterval {
 }
 
 impl BlockInterval {
-    /// Create a new block interval
+    /// Creates a new block interval.
     ///
-    /// A valid block range is of the form:
-    ///     * `block_number` for a single block number
-    ///     * `lhs..rhs`, `lhs..=rhs` as an exclusive/inclusive range
-    ///     * `lhs..` for a range starting from `lhs` to the chain tip. `lhs..=`
-    ///       is also valid format.
+    /// If end_block is None, the interval is unbounded and will follow from
+    /// start_block. If start_block == end_block, the interval is a single
+    /// block. Otherwise, the interval is an inclusive range from start_block to
+    /// end_block.
     ///
-    /// # Example
-    ///
-    /// ```rust
-    ///    # use alloy::rpc::types::eth::BlockId;
-    ///    # use zero::block_interval::BlockInterval;
-    ///    assert_eq!(BlockInterval::new("0..10").unwrap(), BlockInterval::Range(0..10));
-    ///    assert_eq!(BlockInterval::new("0..=10").unwrap(), BlockInterval::Range(0..11));
-    ///    assert_eq!(BlockInterval::new("32141").unwrap(), BlockInterval::SingleBlockId(BlockId::Number(32141.into())));
-    ///    assert_eq!(BlockInterval::new("100..").unwrap(), BlockInterval::FollowFrom{start_block: 100});
-    /// ```
-    pub fn new(s: &str) -> anyhow::Result<BlockInterval> {
```

Comment:
we should close https://github.com/0xPolygonZero/zk_evm/issues/284 if we're dropping all the parsing :)